src/Enable.Extensions.Queuing.Abstractions/BaseQueueClient.cs

```diff
@@ -73,6 +73,49 @@ public void Dispose()
             return EnqueueAsync(messages: batch, cancellationToken: cancellationToken);
         }
 
+        public Task EnqueueAsync(
+            byte[] content,
+            string sessionId,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            IQueueMessage message = new QueueMessage(content, sessionId);
+
+            return EnqueueAsync(message, cancellationToken);
+        }
+
+        public Task EnqueueAsync(
+            string content,
+            string sessionId,
+            CancellationToken cancellationToken = default)
+        {
+            return EnqueueAsync<string>(content, sessionId, cancellationToken);
+        }
+
+        public Task EnqueueAsync<T>(
+            T content,
+            string sessionId,
+            CancellationToken cancellationToken = default)
+        {
+            var message = SerializeQueueMessage(content, sessionId);
+
+            return EnqueueAsync(message, cancellationToken);
+        }
+
+        public Task EnqueueAsync<T>(
```

Comment:
The overload of this method that does not take sessionId could call this overload to avoid duplicated logic.